lib/linguist/generated.rb

```diff
@@ -708,6 +709,16 @@ def generated_jooq?
       lines.first(2).any? { |l| l.include? 'This file is generated by jOOQ.' }
     end
 
+    # Internal: Is this a generated Delphi Interface file for a type library?
+    #
+    # Delphi Type Library Import tool generates *_TLB.pas files based on .ridl files.
+    # They are not meant to be altered by humans.
+    #
+    # Returns true or false
+    def generated_pascal_tlb?
+      name.downcase =~ /_tlb\.pas$/
```

Comment:
Shouldn't we match this case-insensitively?

Reply:
At some point, I thought about that.
In the most cases, the suffix is named with upper cases, however, I found a file (see the link to example source) with lower case. That's why, I decided to switch for both lower and upper cases (as it is now - my assumption here: `.downcase` set all letters in lower case, so I compare with lower case string). Also, the extension itself can be in upper case sometimes.

Reply:
Sorry, bad choice of words. I meant to ask: _"shouldn't we match this using a case-insensitive regex?"_

The shorter, clearer, and conventional way to do this is to use an `/i` modifier:

~~~ruby
name =~ /_tlb\.pas$/i
~~~

Reply:
I see now. Yes, I like your suggestion more. Thank you for a good code review.